account_move_tier_validation/tests/test_tier_validation.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2018 ForgeFlow S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields
+from odoo.exceptions import ValidationError
+from odoo.tests import Form
+from odoo.tests.common import TransactionCase, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestAccountTierValidation(TransactionCase):
+    def test_01_tier_definition_models(self):
+        res = self.env["tier.definition"]._get_tier_validation_model_names()
+        self.assertIn("account.move", res)
+
+    def test_02_form(self):
+        for _type in ("in_invoice", "out_invoice", "in_refund", "out_refund"):
+            self.env["tier.definition"].create(
+                {
+                    "model_id": self.env["ir.model"]
+                    .search([("model", "=", "account.move")])
+                    .id,
+                    "definition_domain": f"[('move_type', '=', '{_type}')]",
+                }
+            )
+            with Form(
+                self.env["account.move"].with_context(default_move_type=_type)
+            ) as form:
+                form.save()
+                self.assertTrue(form.hide_post_button)
+
+    def test_03_move_post(self):
```

Comment:
Odoo is trying to sent emails for the users created in this test.
It can be made more efficient by refactoring it to a setup class and using `DISABLED_MAIL_CONTEXT`

Like done in sale_tier_validation module
https://github.com/OCA/sale-workflow/blob/17.0/sale_tier_validation/tests/test_tier_validation.py#L6-L31